lib/dependencies/AMDDefineDependencyParserPlugin.js

```diff
@@ -40,272 +40,276 @@ class AMDDefineDependencyParserPlugin {
 		this.options = options;
 	}
 
-	newDefineDependency(
-		range,
-		arrayRange,
-		functionRange,
-		objectRange,
-		namedModule
-	) {
-		return new AMDDefineDependency(
-			range,
-			arrayRange,
-			functionRange,
-			objectRange,
-			namedModule
+	apply(parser) {
+		parser.hooks.call.for("define").tap(
+			"AMDDefineDependencyParserPlugin",
+			this.processCallDefine.bind(
+				Object.create(this, {
```

Comment:
This is weird. Just add the `parser` as additional argument to call member methods.